### ISSUE TITLE
[LAB] Week 1_Problem 1_남태웅

### ISSUE THREAD

**Issue by YePpLe** [vlab-kaist/NN101_23S#48]
### Problem

Week 1_Problem 1

### Source Code

```python
import torch
from random import random 
from typing import Callable


##                        Problem 1                           ##
##                                                            ##
##         Arbitrary quadratic function will be given.        ##
## Return the optimal point(global minimum) of given function ##
##          Condition: highest order term is positive         ##
##                  Made by @jangyoujin0917                   ##
##                                                            ##


def solution(func: Callable, start_point: float) -> float: # DO NOT MODIFY FUNCTION NAME    

    alpha = 0.003
    eps = 0.0001

    x = start_point
    h = 10**-9
    for _ in range(10**6):
      dx = (func(x+h) - func(x)) / h
      if abs(dx) <= eps: return x
      x = x - alpha*dx
    return x

if __name__ == "__main__":
    def test_func(x): # function for testing;function for evaluation will be different.
        return x ** 2
    t = 10*random()
    print(solution(test_func, t))
```

### Description

gradient descent
